cmd/containerd-shim-runhcs-v1/service.go

```diff
@@ -432,15 +432,8 @@ func (s *service) DiagStacks(ctx context.Context, req *shimdiag.StacksRequest) (
 		span.AddAttributes(trace.StringAttribute("pod-id", s.tid))
 	}
 
-	buf := make([]byte, 4096)
-	for {
-		buf = buf[:runtime.Stack(buf, true)]
-		if len(buf) < cap(buf) {
-			break
-		}
-		buf = make([]byte, 2*len(buf))
-	}
-	resp := &shimdiag.StacksResponse{Stacks: string(buf)}
+	stacks := debug.Stack()
```

Comment:
`debug.Stack` only returns the stack for the current goroutine, rather than all goroutines (it calls `runtime.Stack` with `all` set to `false`).

Reply:
aaaaaaand that's why

Reply:
well this is tragic lol I should have looked a little harder at their implementation. I'll add it back.